05-Jason-Luo/wk01/5-fri/trainsystem.js

```diff
@@ -0,0 +1,102 @@
+        //Reverse array to get correct order, then splice off 'Union Square' from array, as it is unneccessary
+        backTrip.splice(backTrip[selectUnion], 1);
+        return backTrip;
+    }
+};
+
+function firstTrip(line, station) {
+    const selectLine = trainlines[line];
+    const selectStation = trainlines[line].indexOf(station);
+    const selectUnion = trainlines[line].indexOf("Union Square"); //isolate Union square for QoL
+
+    if (selectStation > selectUnion) {
+        return stationToUnion(selectLine, selectUnion, selectStation)
+    } else {
+        return selectLine.slice(selectStation, selectUnion + 1); //find what stations are in between the two stations
+        // Union Square is included in the first trip, to record total stops.
+    }
+};
+
+function secondTrip(line, station) {
+    const selectLine = trainlines[line];
+    const selectStation = trainlines[line].indexOf(station);
+    const selectUnion = trainlines[line].indexOf("Union Square"); //isolate Union square for QoL
+
+    if (selectStation < selectUnion) {
+        return stationToUnion(selectLine, selectUnion, selectStation)
+    } else {
+        const backTrip = selectLine.slice(selectUnion, selectStation + 1); //find what stations are in between the two stations on the return trip
+        // Slice from selectUnion to selectStation plus 1, to account for being one off
+        backTrip.splice(backTrip[selectUnion], 1); // Splice off Union Square as it is not necessary for second trip log
+        return backTrip;
+    }
+};
+
+function sameLine(line, stationStart, stationEnd) {
+    const selectLine = trainlines[line];
+    const topStation = trainlines[line].indexOf(stationStart);
+    const bottomStation = trainlines[line].indexOf(stationEnd); //isolate Union square for QoL
+
+    if (bottomStation < topStation) {
+        return selectLine.slice(bottomStation, topStation + 1).reverse();
+        //slice from end (bottomStation), then reverse Array to get correct order of Array
+    } else {
+        return selectLine.slice(topStation, bottomStation + 1)
+        //if bottomStation > topStation, return array plus the last stop
+    };
+};
+
+function startEndUnion(lineStart, stationStart, lineEnd, stationEnd) {
+    if (stationStart === "Union Square") {
+        console.log(`You are travelling from ${stationStart} to ${stationEnd}, on the ${lineEnd} line.`);
+    } else {
+        console.log(`You are travelling from ${stationStart} on the ${lineStart} line, to ${stationEnd}.`);
+    }
+};
+
+function planTravel(lineStart, stationStart, lineEnd, stationEnd) {
+    if (lineStart === lineEnd || (stationStart && stationEnd) === "Union Square") {
+        //See if travel is on same line or remaining at Union Square
+
+        if (stationStart === stationEnd) {
+            console.log(`You are staying at the same location. Please select two different stations to travel.`);
+        }; //If traveller is staying at same station
```

Comment:
```suggestion
            console.log(`You are staying at the same location. Please select two different stations to travel.`);
            return;
        }; //If traveller is staying at same station
```
You probably want to end your function here if the stations are the same. This can be done using a `return` statement.